### PR TITLE
Support subdirectories of a git repository as target

### LINKED ISSUE #68: Add support for cruft create/update in a sub directory in an existing repository
cruft Version: 2.5.0

This may come across as a very basic issue, but `cruft update` isn't working as intended.

My steps were:
1) run `cruft create repo-with-cookiecutter-template` within an already existing project
2) git push the newly created project up to the repo
3) make a minor update and push the repo-with-cookiecutter-template
4) run cruft update

This shows that an update occurs, and when I enter "v" I can see a diff (a minor change to a readme)

When I enter "y" the only thing thing that gets update is .cruft.json and the diff is not applied.

Feels like there must be some basic step I'm missing but the docs don't show anything and the closed issues don't seem to have this as an issue.

Any help would be much appreciated.

Thanks

### PR DESCRIPTION
**Pull request by jlec**
Cookie deployments can very well be in a subdirectory of a git repository. cruft has problems to apply patches into subdirectories of a git repo. This PR adds support for this.

Fixes #68

### CODE REVIEW

File: cruft/_commands/update.py [hunk 1]
```diff
@@ -144,8 +144,24 @@ def _apply_patch_with_rejections(diff: str, expanded_dir_path: Path):
...
+                stdout=PIPE,
+                check=True,
+                cwd=expanded_dir_path,
+            )
+            .stdout.decode()
+            .strip()
+        )
+
+        git_apply = ["git", "apply", "-3"]
+        if len(offset) > 0:
```
Comment (samj1912) [inline cruft/_commands/update.py:160]:
```suggestion
        if offset:
```

Comment (samj1912) [reply cruft/_commands/update.py:160]: Also update apply patch with rejections with the offset code?

**Review by samj1912 — CHANGES_REQUESTED** [review]

File: cruft/_commands/update.py (hunk 1 above)
Comment (jlec) [reply cruft/_commands/update.py]: @samj1912 something was odd on this PR. Updated the code.

Comment (jlec) [reply cruft/_commands/update.py:160]: You mean _apply_patch_with_rejections?

Comment (samj1912) [reply cruft/_commands/update.py]: > You mean _apply_patch_with_rejections?

Yup! :)

**Comment by samj1912** [comment]
@jlec any update on this?

**Comment by jlec** [comment]
@samj1912 should be good to go.

**Comment by samj1912** [comment]
It seems like the CI is failing on tests. Can you take a look please?

File: cruft/_commands/update.py (hunk 1 above)
Comment (samj1912) [reply cruft/_commands/update.py]: Can you also make this change please :)

**Comment by mLupine** [comment]
Hi @jlec, do you need any assistance with getting this fixed? I'd really like to have it released as I'm not able to use Cruft without this change, so let me know if I could be of any help so that it could get merged ASAP 🙂

**Comment by jlec** [comment]
> Hi @jlec, do you need any assistance with getting this fixed? I'd really like to have it released as I'm not able to use Cruft without this change, so let me know if I could be of any help so that it could get merged ASAP 🙂

@mLupine Yes please. If you have a quick fix, please let me know.

**Comment by aperiodic** [comment]
@jlec I have fixed the problem with this PR in https://github.com/jlec/cruft/pull/2/commits/c7eb4f1408a2838b2ea9ab6eebf9231d87d00e6f, and opened a PR on this PR's branch. If you merge my PR, that should fix the tests for this PR.

@samj1912 let me know if there's a better process for getting my fix added to this PR than what I described above.

**Comment by samj1912** [comment]
> @jlec I have fixed the problem in [aperiodic@c7eb4f1](https://github.com/aperiodic/cruft/commit/c7eb4f1408a2838b2ea9ab6eebf9231d87d00e6f), and opened [a PR on this PR's branch](https://github.com/jlec/cruft/pull/2). If you merge my PR, that should fix the tests for this PR.
> 
> @samj1912 let me know if there's a better process for getting my fix added to this PR than what I described above.

Thanks - I cherry picked your commit to this pr. Will merge once tests pass.

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
Thanks for this PR! Sorry it took so long to get this merged!

**Comment by sharadpattanshetti** [comment]
Thanks @jlec 
Let me try with manually injecting .cruft.json and see how it works!

**Pull request merged**